src/main/java/seedu/address/model/LoanRecords.java

```diff
@@ -1,44 +1,88 @@
 package seedu.address.model;
 
-import javafx.collections.FXCollections;
-import javafx.collections.ObservableList;
+import static java.util.Objects.requireNonNull;
+
+import java.util.Collection;
+import java.util.HashMap;
+
 import seedu.address.model.loan.Loan;
+import seedu.address.model.loan.LoanId;
 
 /**
- * Wraps all data at the address-book level
- * Duplicates are not allowed (by .isSamePerson comparison)
+ * Wraps all {@code Loan} data at the LoanRecords level.
  */
 public class LoanRecords implements ReadOnlyLoanRecords {
 
-    private ObservableList<Loan> listOfLoans = FXCollections.observableArrayList();
+    private HashMap<LoanId, Loan> loansMap;
 
+    /**
+     * Creates a LoanRecords using the Loans in the {@code toBeCopied}
+     */
     public LoanRecords(ReadOnlyLoanRecords toBeCopied) {
+        requireNonNull(toBeCopied);
 
+        loansMap = new HashMap<>();
+        toBeCopied.getLoanCollection().forEach(loan -> loansMap.put(loan.getLoanId(), loan));
     }
-    /**
-     * Creates an AddressBook using the Persons in the {@code toBeCopied}
-     */
-    public LoanRecords() {
 
+    public LoanRecords() {
+        loansMap = new HashMap<>();
     }
 
+    /**
+     * Get all the Loan objects tracked by LoanRecords.
+     *
+     * @return All the Loans in a {@Code Collection<Loan>}.
+     */
     @Override
-    public ObservableList<Loan> getLoanList() {
-        return FXCollections.unmodifiableObservableList(listOfLoans);
+    public Collection<Loan> getLoanCollection() {
+        return loansMap.values();
     }
 
+    /**
+     * Checks if the LoanRecords contains this {@code loan} object or not.
+     *
+     * @param loan Loan object to be checked.
+     * @return True if LoanRecords contains this {@code loan} object, false otherwise.
+     */
     public boolean hasLoan(Loan loan) {
-        return false;
+        return loansMap.containsValue(loan);
```

Comment:
Would you consider `return hasLoan(loan.getLoadId())` to change the complexity from O(n) to O(1)? Just a minor optimisation considering we are intending for the system to manage many books and loans in the future

Reply:
Great catch, will change!

Reply:
He probably created this extra function signature so that the user can search for a particular loan in the future. If that's so I think it'll be good to just remove this entire function for now since we're taught to forget about possible future requirements.